server/controllers/profile.js

```diff
@@ -0,0 +1,70 @@
+const express = require('express');
+const Profile = require('../models/Profile');
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile
+// @desc Create profile
+// @access Public
+exports.createProfile = asyncHandler(async (req, res, next) => {
+    // Checking for empty input
+    for (let key in req.body) {
+        if (req.body[key].trim() === "") {
+          res.status(406);
+          throw new Error("Invalid input, please do not send empty input(s)")
+        }
+      }
```

Comment:
I have a few notes about this code block:

- I'd say the comment at `line 9` is a bit redundant as the code block is fairly self-explanatory, try to reserve comments for more complicated code blocks or to explain *why* a piece of code is there.
- I suspect you may have slightly misunderstood the `406` status code, to be fair it is one of the more obscure status codes 🤷🏼‍♂️ . You can have a look at [*its mdn doc page*](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406) and/or [*content negotiation*](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation#principles_of_content_negotiation) for more info.
- It's great to see defensive coding strategies here 💯 . But I think we might be able to do this in a better way:
  - First issue is that `body[key]` might not be a `string` so it might not have a `trim` method (see the note [here](https://expressjs.com/en/resources/middleware/body-parser.html)),
  - As @ezeilo-su suggested, we can [destructure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring) the body to explicitly state what input/properties we expect,
  - I believe you implemented the `Profile` model with options such as `trim` and `required`... These options have the power to achieve the same result as you're doing here: `trim` option can trim a string, `required` option can be used to ensure fields are well-defined and aren't empty. Anything that doesn't fit the validation you define there will end up throwing an error so you won't have to do as much validation in the controller body 👍🏼